src/precice/impl/SolverInterfaceImpl.cpp

```diff
@@ -415,8 +415,7 @@ double SolverInterfaceImpl::advance(
     performDataActions({action::Action::WRITE_MAPPING_POST}, time);
   }
 
-  PRECICE_DEBUG("Advance coupling scheme");
-  _couplingScheme->advance();
+  advanceCouplingSchemes();
```

Comment:
```suggestion
  advanceCouplingScheme();
```
? There is only one.
I am also not sure whether we need this function.
Very likely, we might want to do more things in between both exchanges.